Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift

```diff
@@ -397,25 +403,38 @@ public final class PackageInfoMapper: PackageInfoMapping {
         path: AbsolutePath,
         packageFolder: AbsolutePath,
         productTypes: [String: XcodeGraph.Product],
-        productDestinations: [String: XcodeGraph.Destinations],
+        packageSettings: TuistCore.PackageSettings,
         baseSettings: XcodeGraph.Settings,
-        targetSettings: [String: XcodeGraph.SettingsDictionary]
+        targetSettings: [String: XcodeGraph.SettingsDictionary],
+        onlySPMProject: Bool
     ) throws -> ProjectDescription.Target? {
         // Ignores or passes a target based on the `type` and the `packageType`.
         // After that, it assumes that no target is ignored.
         switch target.type {
         case .regular, .system, .macro:
             break
-        case .test, .executable:
+        case .executable:
             switch packageType {
-            case .external:
-                logger.debug("Target \(target.name) of type \(target.type) ignored")
+            case .remote:
+                logger.debug("Target \(target.name) of type \(target.type) is ignored.")
                 return nil
             case .local:
                 break
             }
+        case .test:
+            switch packageType {
+            case .remote:
+                logger.debug("Target \(target.name) of type \(target.type) is ignored.")
+                return nil
+            case .local:
+                if onlySPMProject { break }
+                guard packageSettings.includeLocalPackageTestTargets else {
```

Comment:
What do you think of augmenting the `includeLocalPackageTestTargets` to be true if we are dealing with an SPM only project, so we don't have to pass the property through so many interfaces?

I think it also makes sense conceptually to modify `PackageSettings` to configure the SPM project.

The only trouble is that we don't respect the `includeLocalPackageTestTargets` if it's explicitly set to `false` in an SPM only package project. I think that's fine but we might want to document that.

Reply:
It's amazing advice for me. I really thought about not passing a parameter, but couldn't come up with it. 😮